mne_bids/write.py

```diff
@@ -1339,12 +1339,19 @@ def write_raw_bids(raw, bids_path, events_data=None, event_id=None,
            Symlinks are currently only supported on macOS and Linux. We will
            add support for Windows 10 at a later time.
 
-    empty_room : BIDSPath | None
+    empty_room : mne.io.Raw | BIDSPath | None
```

Comment:
can you explain to me the use of the BIDSPath? I am not sure to understand how this helps

Reply:
BIDSPath is what we supported for quite a while now as the empty_room parameter. The idea here was that when writing experimental data with `write_raw_bids`, you can also pass a BIDSPath pointing to an empty-room recording (that you have manually written beforehand), and we would then add this empty-room recording to the experimental data's sidecar as `AssociatedEmptyRoom`.

Problem with this approach is that you **must** write the empty-room recording first, i.e. before you write the experimental data. This is confusing.

So now, we also allow for users to pass the empty-room raw directly when writing the experimental raw via `write_raw_bids`, essentially reducing the number of required `write_raw_bids` calls from two to just one.

Reply:
👍